src/relay/op/nn/convolution_make.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file src/relay/op/nn/make_convolution.h
+ * \brief utilities for creating convolution ops
+ */
+#ifndef TVM_RELAY_OP_NN_CONVOLUTION_MAKE_H_
+#define TVM_RELAY_OP_NN_CONVOLUTION_MAKE_H_
+
+#include <tvm/relay/attrs/nn.h>
+#include <tvm/relay/op.h>
+
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace tvm {
+namespace relay {
+
+template <typename T>
+Expr MakeConv(Expr data, Expr weight, Array<IndexExpr> strides, Array<IndexExpr> padding,
```

Comment:
mark them as inline since the definition is in the header? Apply to other cases as well

Reply:
:+1: I seem to have a couple of bugs somewhere, chasing down the failing unit tests now.